### PR TITLE
  ROB-3735 Add VictoriaLogs toolset for log search and analysis

### PR DESCRIPTION
**Pull request by naomi-robusta**
## Summary

This PR adds a new VictoriaLogs toolset to HolmesGPT, enabling users to query logs from VictoriaLogs using LogsQL. The toolset provides comprehensive log search, stream discovery, and field enumeration capabilities.

## Key Changes

- **New VictoriaLogs Toolset** (`holmes/plugins/toolsets/victorialogs/victorialogs.py`):
  - `VictoriaLogsConfig`: Configuration class supporting API URL, basic auth, bearer tokens, custom headers, SSL verification, and multi-tenancy (AccountID/ProjectID)
  - `VictoriaLogsToolset`: Main toolset class with health check and request handling
  - Five specialized tools:
    - `VictoriaLogsQuery`: Execute LogsQL queries and return matching log entries
    - `VictoriaLogsStreams`: List log streams (label sets) matching a query
    - `VictoriaLogsFieldNames`: Discover available fields in logs
    - `VictoriaLogsFieldValues`: Enumerate unique field values with hit counts
    - `VictoriaLogsHits`: Time-bucketed log counts for trend analysis
  - Clickable VMUI links for query exploration
  - Comprehensive error handling with detailed diagnostics

- **Documentation** (`docs/data-sources/builtin-toolsets/victorialogs.md`):
  - Configuration examples for no-auth, basic auth, and bearer token scenarios
  - Multi-tenancy setup guide
  - Common use cases and capabilities reference

- **LLM Instructions** (`holmes/plugins/toolsets/victorialogs/instructions.jinja2`):
  - LogsQL syntax cheatsheet for the LLM
  - Tool usage guidance and best practices

- **Comprehensive Tests** (`tests/plugins/toolsets/test_victorialogs.py`):
  - Unit tests for configuration, health checks, and all five tools
  - Authentication tests (basic auth, bearer token, custom headers)
  - Error handling and edge case coverage
  - Live integration tests (optional, gated by `VICTORIALOGS_URL` env var)

- **Integration Tests** (`tests/llm/fixtures/test_ask_holmes/`):
  - Test case 257: Log search with error code extraction (anti-hallucination validation)
  - Test case 258: Stream/service discovery in namespaces

- **Integration Updates**:
  - Registered VictoriaLogsToolset in `holmes/plugins/toolsets/__init__.py`
  - Added to README.md integration list
  - Added to documentation navigation
  - Added pytest markers for regression testing

## Notable Implementation Details

- **Time Range Handling**: Automatic RFC3339 timestamp conversion with sensible defaults (24-hour lookback)
- **JSON Lines Parsing**: Robust JSONL response parsing with error tolerance
- **Request Building**: Form-encoded POST requests with None-value filtering to prevent literal "None" strings
- **Health Checks**: Dedicated `/health` endpoint verification during prerequisites
- **URL Building**: Automatic VMUI link generation for query exploration with proper URL encoding
- **Error Context**: Detailed error messages including request parameters and response bodies for debugging

https://claude.ai/code/session_01B7qL9BFbxBCowj5Kb8uWnY

## Summary by CodeRabbit

* **New Features**
  * Added VictoriaLogs as a supported logs data source with LogsQL query, stream discovery, field inspection, hit-counts, and optional explore links; supports basic auth or bearer token, extra headers, SSL/timeout, and external dashboard URLs.

* **Documentation**
  * Added full VictoriaLogs setup guide, README entry, navigation/index cards, config examples, and a LogsQL cheatsheet with usage scenarios.

* **Tests**
  * Added unit and end-to-end fixtures and live-server tests covering config, health checks, queries, streams, fields, and hits.

* **Chores**
  * CI now injects VICTORIALOGS_URL as a secret-backed env var and added a pytest marker.

### CODE REVIEW

File: holmes/plugins/toolsets/victorialogs/instructions.jinja2 [hunk 1]
```diff
@@ -0,0 +1,24 @@
+VictoriaLogs is a high-performance log database. Query it using LogsQL.
+
+Available tools:
```
Comment (arikalon1) [inline holmes/plugins/toolsets/victorialogs/instructions.jinja2]: I think this is redundant - holmes knows the available tools.

**Review by arikalon1 — DISMISSED** [review]
nice work, left 1 minor comment

**Review by moshemorad — APPROVED** [review]
